### PR TITLE
using the bitstring class for bits

### PR DESCRIPTION
**Pull request by h4ck3rm1k3**
*Bit string support
*handing of null int

### CODE REVIEW

File: parquet/encoding.py [hunk 1]
```diff
@@ -3,21 +3,31 @@
...
 def read_plain_int32(fo):
     """Reads a 32-bit int using the plain encoding"""
-    tup = struct.unpack("<i", fo.read(4))
-    return tup[0]
+    d = fo.read(4)
+    if len(d) > 3:
+        tup = struct.unpack("<i", d)
+        return tup[0]
+    else:
+        return 0
```
Comment (jcrobak) [inline parquet/encoding.py]: is this valid per for the format definition? If so, I'd love to see a unit test.

File: parquet/encoding.py [hunk 2]
```diff
@@ -3,21 +3,31 @@
...
 from ttypes import Type
 
 logger = logging.getLogger("parquet")
 
 
 def read_plain_boolean(fo):
     """Reads a boolean using the plain encoding"""
-    raise NotImplemented
 
+    a = bitstring.Bits(fo)
```
Comment (jcrobak) [inline parquet/encoding.py]: since this is only used in one place... can we achieve the same without the new dep?

File: parquet/encoding.py (hunk 1 above)
Comment (h4ck3rm1k3) [inline parquet/encoding.py]: It has been so long since I reported this, I would have to look at this again.

Comment (jcrobak) [inline parquet/encoding.py]: i'm fairly certain that this is a work-around for https://github.com/jcrobak/parquet-python/issues/24, which I hope to fix soon.

**Comment by jcrobak** [comment]
i'm going to close this out, as I've been looking into refactor this to improve performance. Thanks for the contribution, though! it's been important to uncover a few bugs and help me understand how to improve the library.

**Pull request closed without merging**